rest-api/dao/participant_dao.py

```diff
@@ -114,25 +114,43 @@ def _do_update(self, session, obj, existing_obj):
                             else None)
       need_new_summary = True
 
-    if obj.organizationId or obj.siteId or obj.hpoId:
-      site, organization, awardee = self.get_pairing_level(obj)
-      obj.organizationId = organization
-      obj.siteId = site
-      obj.hpoId = awardee
-      if awardee:
-        # get provider link for hpo_id (awardee)
-        obj.providerLink = make_primary_provider_link_for_id(awardee)
+    update_pairing = True
+    if obj.siteId is None and obj.organizationId is None and obj.hpoId is None and \
+      obj.providerLink == 'null':
```

Comment:
obj.providerLink should not equal 'null'. The fact that it does sometimes is a bug in our code here:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/dao/participant_dao.py#L258

If you fix that, this check would just be obj.providerLink is None.

If you want you can fix that in a follow-up... but we shouldn't keep setting providerLink to 'null'.